.github/workflows/publish.yml

```diff
@@ -0,0 +1,79 @@
+name: publish
+on:
+  push:
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+'
+      - '[0-9]+.[0-9]+.[0-9]+-beta.[0-9]+'
+jobs:
+  pre-publish:
+    name: Extract version
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version_parse.outputs.version }}
+      prerelease: ${{ steps.version_parse.outputs.prerelease }}
+      skip-publish: ${{ steps.version_parse.outputs.skip-publish }}
+    steps:
+      - name: Decode version from tag
+        id: version_regex
+        uses: KyoriPowered/action-regex-match@v3
+        with:
+          text: ${{ github.ref_name }}
+          regex: '^(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)(?:-(beta)\.(?:0|[1-9]\d*))?$'
+      - name: 'Parse version'
```

Comment:
todo: Watch out you put some single quotes around the step name here where there's no quotes on every other steps or jobs name